src/Wellcome.Dds/Wellcome.Dds.Repositories/Presentation/IIIFBuilderParts.cs

```diff
@@ -489,6 +495,52 @@ public void Canvases(Manifest manifest, IManifestation metsManifestation, State
             }
         }
 
+        private static string? GetBetterAVCanvasLabel(IManifestation metsManifestation, IPhysicalFile physicalFile)
+        {
+            string? betterCanvasLabel = null;
+            // Find a better Canvas label for an AV canvas by seeing if this file has one
+            // single corresponding logical struct div. This is very conservative in deciding
+            // what a "structural" logical struct div is. Later, we might want TOCs etc - but they 
+            // can come from normal range-building behaviour.
+            var logicalStructs = metsManifestation.RootStructRange?.Children;
+            if (logicalStructs != null && logicalStructs.HasItems())
```

Comment:
`HasItems()` will handle logicalStructs being `null`

Reply:
yeah, might have been an absent-minded listening to a Rider suggestion, which doesn't seem to work out that it is null checking.

Reply:
Actually the next line requires that it isn't null

Reply:
That should be okay though, it won't enter the if statement if it is null as `HasItems` will be false

Reply:
doh..